app/scripts/modules/core/src/presentation/hooks/useIsMobile.hook.ts

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { throttle } from 'lodash';
+
+import { useEventListener } from './useEventListener.hook';
+import { useLatestCallback } from './useLatestCallback.hook';
+
+const { useState, useCallback } = React;
+
+/////////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////
+///                  WARNING: EXPERIMENTAL                    ///
+/// The details of this implementation (and its API contract) ///
+///          may change in the foreseeable future             ///
+/////////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////
+
+const BREAKPOINT_MOBILE = 1024;
+
+export const useIsMobile = () => {
+  const [isMobile, setIsMobile] = useState(window.innerWidth <= BREAKPOINT_MOBILE);
+
+  const resizeListener = useLatestCallback(() => setIsMobile(window.innerWidth <= BREAKPOINT_MOBILE));
+  const memoizedResizeListener = useCallback(throttle(resizeListener, 200), []);
```

Comment:
Instead of throttling, we could call `setIsMobile` only when the value has changed.  This would eliminate any potential 200ms latency.  The callback would get called a lot, but the logic is fast and it shouldn't matter.

Reply:
yeah you're totally right. I think my past trauma with other things made me a bit too trigger happy with reaching for a full-on throttle. will update.